clowder/kafka.go

```diff
@@ -16,60 +16,86 @@ package clowder
 
 import (
 	"fmt"
+
 	"github.com/RedHatInsights/insights-operator-utils/kafka"
 	api "github.com/redhatinsights/app-common-go/pkg/api/v1"
 )
 
 // Common constants used for logging and error reporting
 const (
-	noBrokerConfig = "warning: no broker configurations found in clowder config"
-	noSaslConfig   = "warning: SASL configuration is missing"
-	noTopicMapping = "warning: no kafka mapping found for topic %s"
+	noOriginalBroker = "warning: no original broker configuration found; aborting"
+	noBrokerConfig   = "warning: no broker configurations found in clowder config"
+	noSaslConfig     = "warning: SASL configuration is missing"
+	noTopicMapping   = "warning: no kafka mapping found for topic %s"
 )
 
 // UseBrokerConfig tries to replace parts of the BrokerConfiguration with the values
-// loaded by Clowder
-func UseBrokerConfig(brokerCfg *kafka.BrokerConfiguration, loadedConfig *api.AppConfig) {
-	// make sure broker(s) are configured in Clowder
+// loaded by Clowder. It expects brokerCfg to already be initialized
+func UseBrokerConfig(brokerCfg kafka.BrokersConfig, loadedConfig *api.AppConfig) kafka.BrokersConfig {
+	numBrokerConfigs := len(brokerCfg)
+	if numBrokerConfigs == 0 {
+		// if original brokers config is totally empty, do nothing.
+		// this shouldn't happen, but we need to control this scenario
+		// to avoid panics.
+		fmt.Println(noOriginalBroker)
```

Comment:
Why not use log instead of `fmt`?

Reply:
My bad. In original code, clowder was being imported before "fmt" package and we couldn't use it. Here we probably can. I'll fix it